app/code/Magento/CatalogSearch/Model/Adapter/Mysql/Aggregation/DataProvider/SelectBuilderForAttributeTypePrice.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider;
+
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\App\ScopeResolverInterface;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Search\Request\BucketInterface;
+use Magento\Store\Model\Store;
+
+/**
+ * Build select for attribute "price".
+ */
+class SelectBuilderForAttributeTypePrice
```

Comment:
From `if ($attribute->getAttributeCode() === 'price') {` it does not look like we deal with attribute type price but rather with price attribute. Please change `ForAttributeTypePrice` to `ForPriceAttribute`.

More general^ why do wee need two classes instead of encapsulating logic in `SelectBuilderForAttribute`?

Reply:
Seems like you are right. I encapsulated logic in `SelectBuilderForAttribute()`.